src/components/scripture/reference.ts

```diff
@@ -99,6 +101,18 @@ const bookIndexFromName = (name: string) => {
   }
   throw new Error('No book matched "' + name + '"');
 };
+const chapterIndexFromChapterNumber = (
+  bookIndex: number,
+  chapterNumber: number
+) => {
+  const chapter = books[bookIndex].chapters.find((c) => {
+    return c === chapterNumber;
+  });
+  if (chapter) {
+    return books[bookIndex].chapters.indexOf(chapter);
+  }
```

Comment:
Is chapter index different from chapter number or is this just validating that the given index is in the array?

Why not skip the `.find()`?
```tsx
const chapterIndex = books[bookIndex].chapters.indexOf(chapter);
if (chapterIndex === -1) {
   throw new InputException('');
}
return chapterIndex;
```

Reply:
Yes, chapter index is different from chapter number.
Ah, can skip the `.find()` .